src/steps/test-projects.ts

```diff
@@ -23,16 +23,28 @@ export default async function testProjects(testProjectPaths: string[], finalResu
 				execa('yarn', [`test`], {
 					cwd: testProjectPath,
 					all: true,
-				}).catch((error) => {
-					// Add to final result
-					finalResult.failedTests.push({
-						project: path.basename(testProjectPath),
-						output: error.all ?? 'No output...',
-					});
+				})
+					.then(
+						printSuccessfulOutput
+							? (result) => {
+									// Add to final result
+									finalResult.successfulTests.push({
+										project: path.basename(testProjectPath),
+										output: result.all ?? 'No output...',
+									});
+							  }
+							: undefined
```

Comment:
I prefer a solution where we don't push them into the array, because even though `push` is not a a slow operation, if a someone was implementing Rugged with many many tests this would significantly increase the time it took for output for no significant benefit on the other side.

With that said, I'm certainly fine with always handing it a call back and then conditionally pushing into the array.